rhino/src/main/java/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -363,14 +363,27 @@ public void delete(String name) {
     @Override
     public void delete(int index) {
         checkNotSealed(null, index);
-        slotMap.remove(null, index);
+        slotMap.compute(null, index, ScriptableObject::checkSlotRemoval);
```

Comment:
Can we already take into consideration here that eventually the delete operation needs to start returning true or false, indicating whether the delete indeed occurred or not? 

I know the 'delete' method of the Scriptable interface has a void return type, but this is wrong from an EcmaScript pov, so it would be good to al least make sure that the internal implementation can already properly handle this